### PR TITLE
[APPS] Add listing of applications

### PR DESCRIPTION
**Pull request by jinroh**
Introduces the `GET /apps/` endpoint.

Added `jsonapi.DataList` to marshal list of `jsonapi.Object`.

### CODE REVIEW

File: web/files/files_test.go [hunk 1]
```diff
@@ -274,7 +274,7 @@ func TestCreateDirConcurrently(t *testing.T) {
 	for i := 0; i < n; i++ {
 		select {
 		case res := <-errs:
-			assert.Equal(t, 409, res.StatusCode)
+			assert.True(t, res.StatusCode == 409 || res.StatusCode == 503)
```
Comment (nono) [inline web/files/files_test.go]: Why this change?

File: web/jsonapi/jsonapi.go [hunk 2]
```diff
@@ -52,6 +52,39 @@ func Data(c *gin.Context, statusCode int, o Object, links *LinksList) {
 	c.Data(statusCode, ContentType, body)
 }
 
+// DataList can be called to send an multiple-value answer with a
+// JSON-API document contains multiple objects.
+func DataList(c *gin.Context, statusCode int, objs []Object) {
```
Comment (nono) [inline web/jsonapi/jsonapi.go]: Like `Data`, we should allow to pass `links *LinksList` in parameters. It will be used for pagination, for example.

File: web/files/files_test.go (hunk 1 above)
Comment (jinroh) [inline web/files/files_test.go]: meh.. sometimes Couch does not handle very well that much concurrent call (100 here) on my laptop and returns 503. I wanted to consider these as "false negative"...

File: web/jsonapi/jsonapi.go [hunk 3]
```diff
@@ -52,6 +52,39 @@ func Data(c *gin.Context, statusCode int, o Object, links *LinksList) {
...
+	var err error
+	objsMarshaled := make([]json.RawMessage, len(objs))
+	for i, o := range objs {
+		var j json.RawMessage
+		j, err = MarshalObject(o)
```
Comment (nono) [inline web/jsonapi/jsonapi.go]: Why not remove the `var err error` at top of the func and `var j json.RawMessage`? And initialize them with:

```go
j, err := MarshalObject(o)
```

Comment (jinroh) [inline web/jsonapi/jsonapi.go]: I thought the linter would cry about that, but I was wrong ...

File: web/jsonapi/jsonapi.go (hunk 2 above)
Comment (jinroh) [inline web/jsonapi/jsonapi.go]: done 8873ad40bf86ed49b6e33851407474ab5301ac05

File: web/jsonapi/jsonapi.go (hunk 3 above)
Comment (jinroh) [inline web/jsonapi/jsonapi.go]: Premature linting is the root of all evil 😄
